config/config.properties

```diff
@@ -97,6 +97,21 @@ pravegaservice.dataLog.impl.name=BOOKKEEPER
 # Default value: HDFS
 # pravegaservice.storage.impl.name=HDFS
 
+
+# Storage layout for Tier 2 storage.
+# Valid values:
+#   CHUNKED_STORAGE - Using ChunkedSegmentStorage.
+#   ROLLING_STORAGE - Using RollingStorage.
+# Default value: ROLLING_STORAGE
+# pravegaservice.storage.layout=ROLLING_STORAGE
+
+# Storage metadata format for Tier 2 storage.
+# Valid values:
+#   TABLE_BASED   - Using ChunkedSegmentStorage.
+#   HEADER_BASED  - Using AsyncStorageWrapper.
+# Default value: HEADER_BASED
+# pravegaservice.storage.metadata.format=HEADER_BASED
```

Comment:
Can we make sure that we throw an error (in the `ServiceConfig` constructor) if this happens. We want to prevent startup for illegal combinations.

Reply:
One possible solution that comes to my mind is having a unified enum that lists all legal combinations. Since we have 2 x 2 and one of those is not allowed, we are left with 3 legal combinations. Would this make things less prone to problems?

Reply:
Even the option - HEADER_BASED  + CHUNKED_STORAGE is really not a "real" option. This is more of an upgrade concern and  should be dealt with independently when we have upgrade implemented.
What do you think?

Reply:
it doesn't look like you need both parameters. What about keeping only one, say `pravegaservice.storage.metadata.format`, and inferring the value of the other?

Reply:
Chunked storage can support either new format (default) or legacy format (during upgrade).
So it is StorageLayoutType that determines whether it is SLTS or not.

Originally I had two separate option because we wanted to support  HEADER_BASED + CHUNKED_STORAGE as an upgrade option. 
However it seems it is better to deal with upgrade path separately without bringing that concept here. https://github.com/pravega/pravega/wiki/PDP-34:-Simplified-Tier-2#supported-options

Reply:
Fixed. I removed pravegaservice.storage.metadata.format as an option.
Now it is implicitly TABLE_BASED when lay out is CHUNKED_STORAGE .
When we have upgrade epic/story then we introduce new property corresponding to upgrade option.